### PR TITLE
Improves the coloring of the search fields throughout the App.

### PR DESCRIPTION
**Pull request by diegoreymendez**
This change was spawned by a chat with @osullivanchris.

This PR improves the coloring of the search fields in dark mode throughout the App.  It changes the background to match the navigation bar color, and it changes the search-query-tex-color to be the default (black for light mode, and white for dark mode).

## Search Field Placeholder Text Coloring

| Before | After |
| --- | --- |
|  |  |
|  |  |

 |

## Search Field Input Text Coloring

| Before | After |
| --- | --- |
|  |  |
|  |  |

## To test:

Test in light and dark mode.

1. Go to any place where you expect a search field.
2. Make sure the color of the search field matches the color of the navigation bar.

## Regression Notes

1. Potential unintended areas of impact

Any search field that doesn't have a navigation bar.  Below is a list of places where the search bar is configured using the changes in this PR.

2. What I did to test those areas of impact (or what existing automated tests I relied on)

I tested each of those fields in-app.

✅ AztecPostViewController.swift: this isn't easy to access because it's a search bar for the media picker in the classic editor.  You need to install the classic editor plugin in your site, create a post, and then edit it using WPiOS and add an image from the  WordPress media library.
✅ BlogListViewController.m: the behavior here seems to be what I'd expect, but I do think it's worth mentioning when you give focus to the search field, the background changes color in a slightly unexpected way (code wise).
✅ BlogSelectorViewController.m: this is the site picker.  There's a divider between the VC title and the search bar that I wasn't able to remove without risking introducing regressions.  This was discussed with both @frosty and @osullivanchris and we agreed it could be left like this for now.
✅ SiteTagsViewController.swift: My Site > Site Settings > Tags.  You need to have at least 1 tag created to see the search field.  There's some divider weirdness worth mentioning here, although it wasn't introduced by my changes.
✅ LanguageSelectorViewController.swift: you need a wp.com site to find this one.  My Site > Site Settings > Language.  Same background changing effect as reported for the blog list.
✅ SiteIconPickerPresenter.swift: My Site > Tap Site Icon > Change Site Icon > "WordPress Media" album
✅ TenorPicker.swift: New post > Add media > Add image > Free GIF Library
✅ StockPhotosPicker.swift: New post > Add media > Add image > Free Photo Library
✅ MediaLibraryViewController.swift: My Site > Media
✅ ParentPageSettingsViewController.swift: My Site > Site Pages > Tap on "..." for an existing page > Set Parent
✅ PluginDirectoryViewController.swift: My Site > Plugins (you'll need a site with plugin support)
✅ AbstractPostListViewController.swift: posts and pages list for a site.
✅ PostSettingsViewController.m: Edit a post > Tap "..." on the top right corner > Post Settings > Set Featured Image > "WordPress Media" album.
❌  PostGeolocationViewController.swift: The search field and the coloring here looks quite broken, but this wasn't introduced by this PR, so it's not actually a regression.
✅ ReaderFollowedSitesViewController.swift: Reader > Tap the gear in the top right corner  > "Followed Sites" tab
✅ ReaderSearchViewController.swift: Reader > Tap the magnifying lens on the top right corner
✅ ThemeBrowserViewController.swift: My Site > Themes
✅ TimeZoneSelectorViewController.swift: My Site > Site Settings > Timezone

3. What automated tests I added (or what prevented me from doing so)

None because this change is not easy to automatically test against.

## PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Review by osullivanchris — CHANGES_REQUESTED** [review]
👋  Hey @diegoreymendez thanks for setting up the PR!

In the after, the text doesn't look white. Didn't we say we'd fix it to the higher contrast colour? Maybe that conversation was after this PR was made.

to do a good review, it would also be good to either get my hands on a build, or see more screenshots of light mode, and other locations where the search bar will change.

**Comment by diegoreymendez** [comment]
@osullivanchris - I've updated the screenshots to better reflect the changes that were done.

- The placeholder text color is now the OS default.  There seems to be a higher contrast in dark mode, but in light mode I'm not sure it's changed much.
- The input text color is now the OS default.  The contrast is higher in both modes since the input color is now black and white.
- The magnifying lens icon is custom, so we have to apply the tint manually to try and match the text color.  In light mode the lens looks slightly darked than the text because of a coloring issue with our default colors vs Apple's.  If it's not too big of an issue I'd suggest we talk about this separately via DM.

**Comment by diegoreymendez** [comment]
Ready for review.  The failing CircleCI test is unrelated and should be fixed soon.

**Comment by osullivanchris** [comment]
Looks great so far, thanks for all the extra effort and discussions. The only details I'm noticing is the text looks a little small. 

I believe it should be Body 17pt https://developer.apple.com/design/human-interface-guidelines/ios/visual-design/typography/

Not sure if we are overwriting the default type to customise and make it smaller, or if we are not using the default type at all.

File: WordPress/Classes/Extensions/Colors and Styles/WPStyleGuide+Search.swift [hunk 1]
```diff
@@ -24,25 +24,25 @@ extension WPStyleGuide {
 
         // We have to manually tint these images, as we want them
         // a different color from the search bar's cursor (which uses `tintColor`)
-        let cancelImage = UIImage(named: "icon-clear-searchfield")?.imageWithTintColor(.neutral(.shade30))
-        let searchImage = UIImage(named: "icon-post-list-search")?.imageWithTintColor(.neutral(.shade30))
+        let cancelImage = UIImage(named: "icon-clear-searchfield")?.withTintColor(.secondaryLabel)
```
Comment (frosty) [inline WordPress/Classes/Extensions/Colors and Styles/WPStyleGuide+Search.swift]: I wonder if we should introduce a new named color reference in UIColor+MurielColors called something like `searchBarItem`?

**Review by frosty — APPROVED** [review]
Changes look good! I ran through the various use cases you mentioned and didn't spot anything other than the existing issues you already highlighted. Just one minor code query, but otherwise :shipit:

**Comment by diegoreymendez** [comment]
@osullivanchris - Adjusted the text size for the placeholder, and the input query in search fields as per your request.  It's now set to font style `body`.

New screenshots below.

| Light | Dark |
| --- | --- |
|  |  |
|  |  |

File: WordPress/Classes/Extensions/Colors and Styles/WPStyleGuide+Search.swift (hunk 1 above)
Comment (diegoreymendez) [reply WordPress/Classes/Extensions/Colors and Styles/WPStyleGuide+Search.swift]: I've addressed this and created some new colors for search field customization.

**Review by frosty — APPROVED** [review]

**Comment by diegoreymendez** [comment]
As reported by @frosty and reviewed by @osullivanchris, I've also fixed some styling issues in a search bar in the reader section of the app.

Before:

Screenshot 2021-04-27 at 16 40 13

After:

| Placeholder | Input Query |
| --- | --- |
|  |  |

**Comment by osullivanchris** [comment]
@diegoreymendez its mostly looking great but I have one more change sorry 🙈 

The text size is now correct. But the height of the input field seems to have increased as well. Maybe it is related. If you look at the screenshots at the top of this PR, the input field is approx 32pt in height. That matches up with the components I have, and with other Apple apps. In the latest screenshots and build I just looked at, the input field looks approx 44pt height.

I wonder if the text has made that happen, or there's some other issue. The original height, with the larger text, would have be perfect.

**Review by osullivanchris — CHANGES_REQUESTED** [review]
forgot to use the 'add review' feature. Only change requested is the input field height as described above.

**Comment by diegoreymendez** [comment]
@osullivanchris - Can I ask you to try again with build 46702?

**Review by osullivanchris — APPROVED** [review]
🙌  Yesss! All issues resolved. Looking great. Thanks for all the extra effort to get it tuned up!!

**Comment by diegoreymendez** [comment]
Thanks folks!  Much appreciated!

**Pull request merged**